queries/futures/useGetFuturesMarkets.ts

```diff
@@ -3,26 +3,37 @@ import { useRecoilValue } from 'recoil';
 import { wei } from '@synthetixio/wei';
 
 import { appReadyState } from 'store/app';
-import { isL2State, networkState, walletAddressState } from 'store/wallet';
+import { isL2State, isWalletConnectedState, networkState } from 'store/wallet';
 
 import Connector from 'containers/Connector';
 import QUERY_KEYS from 'constants/queryKeys';
 import { FuturesMarket } from './types';
+import synthetix from '@synthetixio/contracts-interface';
+import loadProvider from '@synthetixio/providers';
 
 const useGetFuturesMarkets = (options?: UseQueryOptions<FuturesMarket[]>) => {
 	const isAppReady = useRecoilValue(appReadyState);
-	const isL2 = useRecoilValue(isL2State);
 	const network = useRecoilValue(networkState);
-	const walletAddress = useRecoilValue(walletAddressState);
-	const { synthetixjs } = Connector.useContainer();
+	// const { synthetixjs } = Connector.useContainer();
+
+	const provider = loadProvider({
+		networkId: 10,
+		infuraId: process.env.NEXT_PUBLIC_INFURA_PROJECT_ID,
+	});
+
+	const snxjs = synthetix({ provider, networkId: 10, useOvm: true });
+
+	const isWalletConnected = useRecoilValue(isWalletConnectedState);
+	const isL2 = useRecoilValue(isL2State);
+	const isReady = isAppReady && !!snxjs;
```

Comment:
this is the main way to fix these queries - since the Futures markets are only supported on Optimism, the FuturesMarketData contract pull needs to happen from an optimism network - if you're connected to mainnet - this query wiill always throw an error regardless of where it is used because the mainnet synthetix.js has no reference to `FuturesMarketData` contracts. This is the error that gets thrown when using the Connector synthetix provider:

<img width="1578" alt="Screen Shot 2022-04-07 at 3 05 10 PM" src="https://user-images.githubusercontent.com/5998100/162311669-ff5a7347-75de-4a1f-8333-dbf1f46d81bd.png">

Reply:
discussed this with Troy, Ralf, and korede - we worked out a better solution on the futures queries